### PR TITLE
Adds a Tekton Task to create github validation interceptor

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

Along with the task, I added some additional documentation.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior

```

### CODE REVIEW

File: examples/event-interceptors/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Sample Event Interceptors
+
+This folder contains some sample event interceptors:
+
+* [Github Validation](#github-validation)
+
+
+### Github Validation
+
+Validates Github webhook payloads using a supplied webhook secret token. Assumes
```
Comment (vtereso) [inline examples/event-interceptors/README.md]: Nit: Github -> GitHub

File: examples/event-interceptors/gh-validate.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 The Tekton Authors
```
Comment (vtereso) [inline examples/event-interceptors/gh-validate.yaml]: Doesn't this already exist?

Comment (dibyom) [reply examples/event-interceptors/gh-validate.yaml]: Indeed it does

File: examples/event-interceptors/README.md (hunk 1 above)
Comment (dibyom) [reply examples/event-interceptors/README.md]: Added a comment to change this. The task does allow the parameter to be changed.

Comment (dibyom) [reply examples/event-interceptors/README.md]: Is it worth it to have both a task as well as s deployment? A user could choose to use either. Maybe having both and that too in two different places is confusing? What do you think @a-roberts ?

Comment (a-roberts) [reply examples/event-interceptors/README.md]: I like the idea of a Tekton Task yeah, users get to see the logs e.g. through the dashboard or kubectl trivially, they have a "record" (in the form of a TaskRun CRD) of the interceptor being created (which might be useful on a prod system where serviceability is an important factor) and it serves as another example of how a good Tekton Task definition should look, so I'm all for having both

File: docs/create-gh-validate-interceptor.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+spec:
+  inputs:
+    params:
+      - name: Name
+        description: "The name of the interceptor Service."
+      - name: GitHubSecretName
+        description: "The Secret name for GitHub webhook secret. This is always mounted and must exist"
+      - name: GitHubSecretStringKey
+        description: "Key name within the Secret for the Github webhook secret."
+      - name: ContainerImage
```
Comment (a-roberts) [inline docs/create-gh-validate-interceptor.yaml]: I'm following these docs implementing a validator for our move to use Triggers with our Tekton Webhooks Extension, it's not clear what this container image might do exactly or if this should be replaced to use, say, my own Docker registry if I go on to ko apply the provided Task

Should the container image, say, run some script perhaps that would return a non-zero exit code if some condition fails? 

What would say script look like (I see https://github.com/tektoncd/triggers/blob/6e81d0f3dea0497f70e32cd5dc30285703d451e7/examples/event-interceptors/github-validate.yaml, but I'm used to validation being done through the Task itself like with https://github.com/tektoncd/experimental/pull/258/files#diff-16e2249a36211431f525dd0f47c1703c)

I know this is a WIP so perhaps that's coming or elsewhere!

The use cases we're looking at specifically include:
- Checking valid X-GitHub signature (checking secretToken used in the validation service with what's set on the webhook)
- Checking a repository URL matches an input parameter (so templates/bindings only get used to trigger PipelineRuns/PipelineResource generation for a given repository, not everything we have webhooks for)
- Eventually checking the type of event coming in (e.g. a push or pull request being created, not just anything)

File: examples/event-interceptors/README.md (hunk 1 above)
Comment (a-roberts) [reply examples/event-interceptors/README.md]: An example TaskRun would be super useful here too btw

File: docs/create-gh-validate-interceptor.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          spec:
+            type: ClusterIP
+            selector:
+              app: $(inputs.params.Name)
+            ports:
+              - protocol: TCP
+                port: 80
+                targetPort: 8080
+          EOF
+          fi
```
Comment (a-roberts) [inline docs/create-gh-validate-interceptor.yaml]: I'm pretty sure the fi's cause errors (we don't have any ifs after all)

```
adams-mbp:triggers_prototype aroberts$ k logs github-validate-run-89mhl-pod-195926 --all-containers
{"level":"warn","ts":1570624241.7803333,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"ref: refs/heads/master\" is not a valid GitHub commit ID"}
{"level":"info","ts":1570624241.781536,"logger":"fallback-logger","caller":"creds-init/main.go:40","msg":"Credentials initialized."}
service/my-interceptor-service created
sh: syntax error: unexpected "fi"
```

Same with the other

File: docs/create-gh-validate-interceptor.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+              matchLabels:
+                app: $(inputs.params.Name)
+            template:
+              metadata:
+                labels:
+                  app: $(inputs.params.Name)
+              spec:
+                containers:
+                  - name: validate
+                    image: $(input.params.ContainerImage)
```
Comment (a-roberts) [inline docs/create-gh-validate-interceptor.yaml]: inputs* everywhere

File: docs/create-gh-validate-interceptor.yaml (hunk 3 above)
Comment (dibyom) [reply docs/create-gh-validate-interceptor.yaml]: So, I tried adding some docs in https://github.com/tektoncd/triggers/blob/6e81d0f3dea0497f70e32cd5dc30285703d451e7/examples/event-interceptors/README.md

but yeah having the task in docs/ and the actual container code in examples/ looks confusing. Open to ideas here. The other tasks are in docs/ so I was following the pattern. Maybe we should have a folder for these tasks somewhere?

**Comment by vtereso** [comment]
Closing as per https://github.com/tektoncd/triggers/pull/160.

**Pull request closed without merging**